job-service-contract/src/main/resources/com/hpe/caf/services/job/swagger.yaml

```
@@ -415,6 +417,56 @@ definitions:
         example:
           "tag:4": "4"
           owner: "bob"
+  expirationPolicy:
+    type: object
+    properties:
+      "Active":
+        $ref: "#/definitions/policy"
+      "Completed":
+        $ref: "#/definitions/policy"
+      "Failed":
+        $ref: "#/definitions/policy"
+      "Cancelled":
+        $ref: "#/definitions/policy"
+      "Waiting":
+        $ref: "#/definitions/policy"
+      "Paused":
+        $ref: "#/definitions/policy"
+      "Expired":
+        $ref: "#/definitions/deletePolicy"
+      "Default":
+        $ref: "#/definitions/policy"
+    description: The expiration policy to be applied on the job
+  deletePolicy:
+    type: object
+    properties:
+      expiryTime:
+        $ref: "#/definitions/expiryTimePattern"
+      expirationOperation:
+        type: string
+        enum:
+          - delete
+    description: The expiration details to be applied on the job
+  expiryTimePattern:
+    type: string
+    pattern: '^(?<fullyear>\d{4})-(?<month>0[1-9]|1[0-2])-(?<mday>0[1-9]|[12][0-9]|3[01])T(?<hour>[01][0-9]|2[0-3]):(?<minute>[0-5][0-9]):(?<second>[0-5][0-9]|60)(?<secfrac>\.[0-9]+)?(Z|(\+|-)(?<offset_hour>[01][0-9]|2[0-3]):(?<offset_minute>[0-5][0-9]))|(((lastUpdateTime)|(createTime))(\+P)[1-9]\d*[DYHM])|(no expiry)$'
```

If we're going to have a token for no expiry I'd rather that it is "none":
```suggestion
    pattern: '^(?<fullyear>\d{4})-(?<month>0[1-9]|1[0-2])-(?<mday>0[1-9]|[12][0-9]|3[01])T(?<hour>[01][0-9]|2[0-3]):(?<minute>[0-5][0-9]):(?<second>[0-5][0-9]|60)(?<secfrac>\.[0-9]+)?(Z|(\+|-)(?<offset_hour>[01][0-9]|2[0-3]):(?<offset_minute>[0-5][0-9]))|(((lastUpdateTime)|(createTime))(\+P)[1-9]\d*[DYHM])|(none)$'
```

---

Done and fixed the issues about extra characters before or after the expression